libethereum/Client.h

```diff
@@ -220,8 +220,8 @@ class Client: public ClientBase, protected Worker
         boost::filesystem::path const& _snapshotPath, WithExisting _forceAction, u256 _networkId);
 
     /// InterfaceStub methods
-    BlockChain& bc() override { return m_bc; }
-    BlockChain const& bc() const override { return m_bc; }
+    std::shared_ptr<BlockChain> bc() override { return m_bc; }
```

Comment:
This should keep returning reference. Otherwise you bump ref count on every access.

Reply:
I'll try to fix this one and similar ones.